tools/rosunit/src/rosunit/xmlrunner.py

```diff
@@ -268,6 +268,8 @@ def run(self, test):
 
         result.print_report_text(sys.stdout, time_taken, out_s, err_s)
 
+        stream.flush()
```

Comment:
Can you please move this line a few lines up - to be directly after the `result.print_report(stream, ...)` call.

(I would have done the change myself but you looks like you unchecked the option to allow maintainers of the target repo to make changes to your PR.)